staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/raw.go

```diff
@@ -44,6 +45,23 @@ var rawTypeTranscodeFuncs = map[reflect.Type]func(reflect.Value) error{
 		re.Raw = j
 		return nil
 	},
+	reflect.TypeFor[metav1.FieldsV1](): func(rv reflect.Value) error {
+		if !rv.CanAddr() {
+			return nil
+		}
+		fields := rv.Addr().Interface().(*metav1.FieldsV1)
+		if fields.Raw == nil {
+			// When Raw is nil it encodes to null. Don't change nil Raw values during
+			// transcoding, they would have unmarshalled from JSON as nil too.
+			return nil
+		}
+		j, err := fields.MarshalJSON()
+		if err != nil {
+			return fmt.Errorf("failed to transcode FieldsV1 to JSON: %w", err)
+		}
+		fields.Raw = j
```

Comment:
so thinking through a GET for a CR, this
1. reads CBOR from storage, the kube-apiserver decode will convert to JSON
2. when encoding response to the client, the kube-apiserver will encode into CBOR
3. when the client decodes the response, the client will convert to JSON
4. the client will probably never introspect this value

is that about right?  For clients that do introspect the value, how would we suggest they do that?

Reply:
That looks right.

> For clients that do introspect the value, how would we suggest they do that so we can later avoid this automated transcode?

Those clients could convert to unstructured. If FieldsV1 implements `UnstructuredConverter`, that conversion can happen directly from either encoding.

Reply:
Does it make sense to add a beta requirement that eliminates the transcoding for my step 1 to prevent decoding until it's demanded?  That would ensure we build the deferred option and keep the load in kube-apiserver lower, correct?

Reply:
Yes, there should be a beta requirement to opt out of the transcoding for kube components. I added that to the tracker issue.

I think the extra kube-apiserver load will come from the other direction (decoding from CBOR requests and not from storage). Initially, kube-apiserver will be transcoding FieldsV1 to JSON when they come from CBOR clients, and field management updates will continue to set JSON, so stored objects should still have JSON FieldsV1 values.

The field manager code uses json-iterator to traverse the trie without unmarshalling it. We probably need CBOR support for something similar in order to avoid unnecessary transcoding and have comparable performance there.